packages/build/src/services/page.ts

```diff
@@ -0,0 +1,104 @@
+import { LogEvent, OutputServiceEvents, ServiceEventBus } from './'
+import {
+  BackendPageBundleService,
+  BackendPageBundleServiceEvents,
+  BrowserPageBundleService,
+  BrowserPageBundleServiceEvents,
+  BundleBackendPageErrorEvent,
+  BundleBackendPageEvent,
+  BundleBrowserPageErrorEvent,
+  BundleBrowserPageEvent,
+  BundledBackendPageEvent,
+  BundledBrowserPageEvent,
+} from './bundler'
+
+export const BundlePageEvent = Symbol('Start page bundling')
+export const BundledPageEvent = Symbol('page bundling completed')
+export const BundlePageErrorEvent = Symbol('Error page bundling')
+export interface PageBundleServiceEvents extends OutputServiceEvents {
+  [BundlePageEvent]: PageBundleService['handleBundling']
+  [BundledPageEvent]: (page: string, time: [number, number]) => void
+  [BundlePageErrorEvent]: () => void
+}
+
+type BrowserPageBundlerBus = ServiceEventBus<BrowserPageBundleServiceEvents>
+type BackendPageBundlerBus = ServiceEventBus<BackendPageBundleServiceEvents>
+
+export class PageBundleService {
+  bus!: ServiceEventBus<PageBundleServiceEvents>
+  bundlerBus!: ServiceEventBus<
+    BrowserPageBundleServiceEvents & BackendPageBundleServiceEvents
+  >
+  browserPageBundler!: BrowserPageBundleService
+  backendPageBundler!: BackendPageBundleService
+  bundled = new Set<'browser' | 'backend'>()
+  bundlingTimer: [number, number] = [0, 0]
+
+  config = {
+    compatible: false,
+    excludes: ['react', 'react-dom', 'react-helmet', 'lodash', 'typestyle'],
```

Comment:
can we put the excludes somewhere global? 
- added packages in in the future will be bundled in case the line is not known to the one who adds it
- these excludes have a duplicate definition in browser_page.ts

Reply:
duplication: fixed

we dont really need to configure this atm. would add this functionality when the need arises

Reply:
agree to disagree :)